R/time_bins.R

```diff
@@ -0,0 +1,206 @@
+#'
+#' #Using a range of intervals defined by two named intervals and equal duration bins
+#' time_bins(interval = c("Fortunian", "Meghalayan"), size = 10, plot = TRUE)
+#'
+#' #Assign bins based on given age estimates
+#' time_bins(interval = c("Fortunian", "Meghalayan"), assign = c(232, 167, 33), plot = TRUE)
+#' @export
+time_bins <- function(interval = c("Fortunian", "Meghalayan"), size = NULL, assign = NULL, scale = "GTS2020", plot = FALSE){
+  #error handling
+  if (!is.character(interval) & !is.numeric(interval)) {
+    stop("`interval` must be of class 'character' or 'numeric'")
+  }
+
+  if (is.numeric(size) == FALSE & is.null(size) == FALSE) {
+    stop("Size should be a numeric or NULL")
+  }
+
+  if (is.logical(plot) == FALSE) {
+    stop("plot should be logical (TRUE/FALSE)")
+  }
+
+  if (is.numeric(assign) == TRUE & any(assign < 0) == TRUE) {
+    stop("Age estimates should be non-negative values.
+  You can transform your data using abs().")
+  }
+
+  if (scale != "GTS2012" & scale != "GTS2020") {
+    stop("Scale must be either 'GTS2012' or 'GTS2020'")
+  }
+
+  #grab data
+
+  #which geological timescale to use?
+
+  if(scale == "GTS2020"){df <- palaeoverse::GTS2020}
+  if(scale == "GTS2012"){df <- palaeoverse::GTS2012}
+
+  #which rank is required? Non-stage level is only available for character string.
+  if(is.character(interval)){
+    rnk <- df[which(df$interval_name %in% interval), c("rank")]
+
+    if(length(rnk) != length(interval)){stop(paste0("Checking spelling of specified intervals.
+    Available intervals are accessible via GTS2020$interval_name and GTS2012$interval_name."))}
+
+    rnk <- unique(rnk)}else{rnk <- "Stage"}
+
+  if(length(rnk) != 1){stop("Interval ranks should be the same, e.g. specifiy only stage or period names, not a combination.")}
+
+  #subset to rank
+  df <- subset(df, rank == rnk)
+
+  #character string entered
+  if(is.character(interval)){
+    if(length(interval) == 1){
+      w <- which(df$interval_name %in% interval)
+      if(length(w) != length(interval)){stop(paste0("Checking spelling of specified intervals.
+    Available intervals are accessible via GTS2020$interval_name."))}
+      df <- df[w,]
+    }
+    if(length(interval) == 2){
+      w <- which(df$interval_name %in% interval)
+      df <- df[w[1]:w[2],]
+    }
+  }
+  #numeric ages entered
+  if(is.numeric(interval)){
+    if(length(interval) == 1){
+      if(interval > max(df$max_ma) | interval < min(df$min_ma)){
+        stop("Value does not appear in range of available intervals: 0 to 541")
+      }
+      df <- df[which.min(abs(df$mid_ma - interval)),]
+    }
+
+    if(length(interval) == 2){
+      max_int <- max(interval)
+      min_int <- min(interval)
+
+      if(max_int > max(df$max_ma) | min_int < min(df$min_ma)){
+        stop("Values do not appear in range of available intervals: 0 to 541")
+      }
+
+      df <- df[which.min(abs(df$min_ma - min_int)):which.min(abs(df$max_ma - max_int)),]
```

Comment:
as above, should change to 
      int_index <- which(min_int < df$max_ma & max_int >= df$min_ma)
      df <- df[int_index,]

Reply:
Implemented.